test.bash

```diff
@@ -45,23 +36,43 @@ if [ "$1" == "docker-full" ]; then
   cond_redirect docker run --rm --name "destructive-tests" -i openhabian/bats-openhabian bash -c 'bats --tap --recursive --filter "destructive-." .'
   cond_redirect echo_process "Test complete, please review result in terminal."
   exit 0
-elif [ "$1" == "shellcheck" ]; then
+elif [[ $1 == "shellcheck" ]]; then
   shellcheck -x -s bash openhabian-setup.sh
   shellcheck -x -s bash functions/*.bash
   shellcheck -x -s bash build-image/*.bash
-elif [ "$1" == "travis" ]; then
+elif [[ $1 == "travis" ]]; then
   # prepare configuration for tests, select debug level here:
   sed -i 's#debugmode=.*$#debugmode=on#' build-image/openhabian.conf
   #sed -i 's#debugmode=.*$#debugmode=maximum#' build-image/openhabian.conf
   if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then
     echo "Cloning openHABian repo from https://github.com/${TRAVIS_PULL_REQUEST_SLUG}.git"
-    sed -i 's#repositoryurl=.*$#repositoryurl=https://github.com/'${TRAVIS_PULL_REQUEST_SLUG}'.git#' build-image/openhabian.conf
-    sed -i 's#clonebranch=.*$#clonebranch='${TRAVIS_PULL_REQUEST_BRANCH}'#' build-image/openhabian.conf
+    sed -i 's|repositoryurl=.*$|repositoryurl=https://github.com/'"${TRAVIS_PULL_REQUEST_SLUG}"'.git|' build-image/openhabian.conf
+    sed -i 's|clonebranch=.*$|clonebranch='"${TRAVIS_PULL_REQUEST_BRANCH}"'|' build-image/openhabian.conf
   else
     echo "Cloning openHABian repo from https://github.com/${TRAVIS_REPO_SLUG}.git"
-    sed -i 's#repositoryurl=.*$#repositoryurl=https://github.com/'${TRAVIS_REPO_SLUG}'.git#' build-image/openhabian.conf
-    sed -i 's#clonebranch=.*$#clonebranch='${TRAVIS_BRANCH}'#' build-image/openhabian.conf
+    sed -i 's|repositoryurl=.*$|repositoryurl=https://github.com/'"${TRAVIS_REPO_SLUG}"'.git|' build-image/openhabian.conf
+    sed -i 's|clonebranch=.*$|clonebranch='"${TRAVIS_BRANCH}"'|' build-image/openhabian.conf
+  fi
+elif [[ $1 == "github" ]]; then
+  repoURL="$(git remote get-url origin)"
+  repoBranch="$(git rev-parse --abbrev-ref HEAD)"
+
+  if ! [[ $repoURL == "https"* ]]; then
+    # Convert URL from SSH to HTTPS
+    username=$(echo "$repoURL" | sed -Ene's#git@github.com:([^/]*)/(.*).git#\1#p')
+    if [[ -z $username ]]; then
+      echo_process "Could not identify git user while converting to SSH URL. Exiting."
+      exit 1
+    fi
+    reponame=$(echo "$repoURL" | sed -Ene's#git@github.com:([^/]*)/(.*).git#\2#p')
+    if [[ -z $reponame ]]; then
+      echo_process "Could not identify git repo while converting to SSH URL. Exiting."
+      exit 1
+    fi
+    repoURL="https://github.com/${username}/${reponame}.git"
   fi
+  sed -i 's|repositoryurl=.*$|repositoryurl='"${repoURL}"'|' build-image/openhabian.conf
```

Comment:
Hm, this is actually not a test, but switching to the current branch/repo.
We will get out an image which tries to read from this branch and not from master - a bit like the git-dev option before.
@mstormi can you please have a look?

Reply:
This serves the same purpose as the Travis option, to setup the environment correctly on PRs. If you would like we could relocate these to a specific `ci-setup.bash` file.

Reply:
Also, it uses whatever the current branch is so if we build from master branch its not an issue.

Reply:
I saw ci-setup.bash takes care not to set repo/branch when it's a PR, that's ok.
But how exactly would we trigger building the image from master or stable then ?

Reply:
It sets it the same way on all triggers, I know this because the builds on my repo work completely fine. All you need to do is push the commit to stable and it should trigger a build with everything properly set.